File: java-2022-prelim/P06.java
import java.util.Scanner;

public class P06 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double m = scanner.nextDouble();
        double c = scanner.nextDouble();
        double delta = (m + 2) * (m + 2) + 4 * (c + 3);
        if (delta < 0) {
            System.out.println("No intersection");
            return;
        }
        double center = m / 2 + 1;
        if (delta == 0) {
            System.out.println("One intersection");
            System.out.printf("%.7f, %.7f", center, m * center + c);
            return;
        }
        double difference = Math.sqrt(delta) / 2;
        double x1 = center + difference;
        double x2 = center - difference;
        System.out.println("Two intersections");
        System.out.printf("%.7f, %.7f%n", x1, m * x1 + c);
        System.out.printf("%.7f, %.7f", x2, m * x2 + c);
    }
}

File: cpp-2022-prelim/p06.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    double m, c;
    cin >> m >> c;
    const double delta = (m + 2) * (m + 2) + 4 * (c + 3);
    if (delta < 0) {
        cout << "No intersection";
        return 0;
    }
    cout << fixed << setprecision(7);
    const double center = m / 2 + 1;
    if (delta == 0) {
        cout << "One intersection" << endl;
        cout << center << ", " << m * center + c;
        return 0;
    }
    cout << "Two intersections" << endl;
    const double difference = sqrt(delta) / 2;
    const double x1 = center + difference;
    const double x2 = center - difference;
    cout << x1 << ", " << m * x1 + c << endl;
    cout << x2 << ", " << m * x2 + c;
}
